desc/geometry/curve.py

```diff
@@ -705,8 +706,24 @@ def __init__(
             errorif(knots[-1] > 2 * np.pi, ValueError, "knots must lie in [0, 2pi]")
             knots = knots[:-1] if closed_flag else knots
 
+        if discontinuous_knots is None:
+            interval_indices = [[0, -1]]
```

Comment:
might be better to default to an empty array, and then in the compute function do something like 

```python
if len(interval_indices):
    # do the piecewise stuff
else:
    # just do regular interpolation over whole interval with period=2pi
```

Reply:
something like the most recent push?